include/sampleflow/producers/differential_evaluation_mh.h

```diff
@@ -0,0 +1,156 @@
+         *     divided by the probability of the transition $\tilde x\to x$.
+         *     Specifically, if the proposal distribution is given by
+         *     $\pi_\text{proposal}(\tilde x|x)$, then the second element
+         *     of the pair returned by the `perturb` argument is
+         *     $\frac{\pi_\text{proposal}(\tilde x|x)}
+         *           {\pi_\text{proposal}(x|\tilde x)}$.
+         *   </ol>
+         *   If samples are from some continuous space, say
+         *   ${\mathbb R}^n$, then the perturbation function is often
+         *   implemented by choosing $\tilde x$ from a neighborhood
+         *   of $x$. If, for example, $\tilde x$ is chosen with a probability
+         *   that only depends on the distance $\|\tilde x-\x\|$ as is often
+         *   done (e.g., uniformly in a disk of a certain radius around
+         *   $x$, or using a Gaussian probability distribution centered at
+         *   $x$), then the ratio returned as second argument is
+         *   $\frac{\pi_\text{proposal}(\tilde x|x)}
+         *           {\pi_\text{proposal}(x|\tilde x)}=1$.
+         * @param[in] n_samples The number of (new) samples to be produced
+         *   by this function. This is also the number of times the
+         *   signal is called that notifies Consumer objects that a new
+         *   sample is available.
+         */
+        void
+        sample (const OutputType &starting_point,
+                const std::function<double (const OutputType &)> &log_likelihood,
+                const std::function<std::pair<OutputType,double> (const OutputType &)> &perturb,
+                const types::sample_index n_samples);
+    };
+
+
+    template <typename OutputType>
+    void
+    DifferentialEvaluationMetropolisHastings<OutputType>::
+    sample (const OutputType &starting_point,
+            const std::function<double (const OutputType &)> &log_likelihood,
+            const std::function<std::pair<OutputType,double> (const OutputType &)> &perturb,
+            const types::sample_index n_samples)
+    {
+      // Make sure the flush_consumers() function is called at any point
+      // where we exit the current function.
+      Utilities::ScopeExit scope_exit ([this]()
+      {
+        this->flush_consumers();
+      });
+
+      std::mt19937 rng;
+      std::uniform_real_distribution<> uniform_distribution(0,1);
+
+      OutputType current_sample         = starting_point;
+      double     current_log_likelihood = log_likelihood (current_sample);
+
+
+      // Loop over the desired number of samples
+      for (types::sample_index i=0; i<n_samples; ++i)
+        {
+          // Loop over the desired number of chains
+          for (unsigned int chain = 0; chain < sample_chains; ++chain)
+            {
+              // Initialize a bool to store whether a sample is accepted
+              bool accepted_sample = false;
+              // Obtain a new sample by perturbation and evaluate its log likelihood
```

Comment:
https://en.cppreference.com/w/cpp/numeric/random/uniform_int_distribution

Use this to determine which chains to use to get crossover samples